internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/fastclient/utils/AbstractClientEndToEndSetup.java

```diff
@@ -496,29 +498,40 @@ protected AvroSpecificStoreClient<String, TestValueSchema> getSpecificThinClient
             .setSslFactory(SslUtils.getVeniceLocalSslFactory()));
   }
 
-  protected void validateMetrics(
+  protected void validateSingleGetMetrics(MetricsRepository metricsRepository, boolean retryEnabled) {
+    validateBatchGetMetrics(metricsRepository, false, 0, 0, retryEnabled);
+  }
+
+  protected void validateBatchGetMetrics(
       MetricsRepository metricsRepository,
       boolean useStreamingBatchGetAsDefault,
       int expectedBatchGetKeySizeMetricsCount,
-      int expectedBatchGetKeySizeSuccessMetricsCount) {
-    String metricPrefix = useStreamingBatchGetAsDefault ? "--multiget_" : "--";
+      int expectedBatchGetKeySizeSuccessMetricsCount,
+      boolean retryEnabled) {
+    String metricPrefix = "." + storeName + (useStreamingBatchGetAsDefault ? "--multiget_" : "--");
     double keyCount = useStreamingBatchGetAsDefault ? expectedBatchGetKeySizeMetricsCount : 1;
     double successKeyCount = useStreamingBatchGetAsDefault ? expectedBatchGetKeySizeSuccessMetricsCount : 1;
     Map<String, ? extends Metric> metrics = metricsRepository.metrics();
+
     // counters are incremented in an async manner, so adding non-deterministic wait
-    TestUtils.waitForNonDeterministicAssertion(10, TimeUnit.SECONDS, () -> {
+    TestUtils.waitForNonDeterministicAssertion(5, TimeUnit.SECONDS, () -> {
+      assertTrue(metrics.get(metricPrefix + "request.OccurrenceRate").value() > 0);
+      assertTrue(metrics.get(metricPrefix + "healthy_request.OccurrenceRate").value() > 0);
+      assertTrue(metrics.get(metricPrefix + "healthy_request_latency.Avg").value() > 0);
+      assertFalse(metrics.get(metricPrefix + "unhealthy_request.OccurrenceRate").value() > 0);
```

Comment:
I see CI failures on this assertion... not sure what's the right way to go about it... should we expect zero unhealthy requests? Or is it acceptable to have some low rate of unhealthy requests? Is there a retry mechanism where some request could fail, get retried, and then succeed? And if that happens, would it get counted here (even though things worked out in the end)?